backend/moonstreamapi/providers/moonworm_provider.py

```diff
@@ -0,0 +1,453 @@
+from dataclasses import dataclass, field
+import logging
+from typing import cast, Dict, Any, List, Optional, Tuple
+
+from bugout.app import Bugout
+from bugout.data import BugoutResource
+
+from moonstreamdb.blockchain import (
+    get_label_model,
+    AvailableBlockchainType,
+)
+from sqlalchemy import or_, and_, text
+from sqlalchemy.orm import Session, Query, query_expression
+from sqlalchemy.sql.expression import label
+
+
+from .. import data
+from ..stream_boundaries import validate_stream_boundary
+from ..stream_queries import StreamQuery
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.WARN)
+
+
+ethereum_event_type = "ethereum_blockchain"
+polygon_event_type = "polygon_blockchain"
+allowed_tags = ["tag:erc721"]
+
+description = f"""Event provider for transactions from the Ethereum blockchain.
+
+To restrict your queries to this provider, add a filter of \"type:{ethereum_event_type}\{polygon_event_type}\" to your query (query parameter: \"q\") on the /streams endpoint."""
+
+default_time_interval_seconds: int = 5 * 60
+
+# 200 transactions per block, 4 blocks per minute.
+estimated_events_per_time_interval: float = 5 * 800
+
+
+@dataclass
+class ArgsFilters:
+    name: str
+    value: Any
+    type: str
+
+
+@dataclass
+class LabelsFilters:
+
+    name: str
+    type: str
+    args: List[ArgsFilters] = field(default_factory=list)
+
+
+@dataclass
+class AddressFilters:
+
+    address: str
+    labels: List[LabelsFilters] = field(default_factory=list)
+
+
+@dataclass
+class Filters:
+
+    addresses: List[AddressFilters] = field(default_factory=list)
+
+
+def python_type(expected_type: str) -> Any:
+    if expected_type.startswith("int"):
```

Comment:
I don't like the use of `.startswith` here. It could give unexpected results in future. I would do something like this instead:

```
if expected_type.lower() == "int" or expected_type.lower() == "integer":
```